### ISSUE TITLE
Feature Request: Making security group user interface consistent

### ISSUE THREAD

**Issue by norman-abramovitz** [cloudfoundry/cli#1171]
The user interface for binding security groups seems inconsistent with the various styles of bind commands.

The platform-wide security group binding uses

  bind-running-security-group <sgn>
  bind-staging-security-group <sgn>

The new interface that uses --lifecycle parameters uses

  bind-security-group --lifecycle <running|staging> <sgn>

You could simplify the interface using either style, but you do not need both forms.   One interface could be deprecated.

Proposed interface 1 

bind-running-security-group <sgn>  [-o org]  [-s space]
bind-staging-security-group <sgn>    [-o org]  [-s space]
unbind-running-security-group <sgn>  [-o org]  [-s space]
unbind-staging-security-group <sgn>    [-o org]  [-s space]
staging-security-groups  [-o org]  [-s space]
running-security-groups [-o org]  [-s space]

Proposed interface 2
bind-security-group <sgn>  --lifecycle <running|staging> [--platform]  [-o org]  [-s space]
unbind-security-group <sgn>  --lifecycle <running|staging>[--platform]  [-o org]  [-s space]
security-groups  --lifecycle <running|staging> [--platform] [-o org]  [-s space]
The lifecycle parameter could become a keyword verses a parameter as well.

**Comment by dkoper** [comment]
Hi @norman-abramovitz 

I would have liked to go for something like interface 2, i.e. a small set of commands with flags indicating their scope. Unfortunately we inherited the current set of commands to which it wasn't trivial to add the space scoped staging ASGs.

Application Security Groups (ASG) are resources that may be replaced by the container to container networking project.

So this time, rather than introducing multiple ways to do the same thing with a consistent interface and having to leave the "old" way around for an indefinite time until we major version bump, I only introduced the extension with --lifecycle and will wait a bit longer to see whether we should tidy this up in the next major version bump, or possibly do away with them altogether.

Regards,
Dies Koper
CF CLI PM
